apps/src/code-studio/components/header/HeaderFinish.jsx

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import $ from 'jquery';
+
+const styles = {
+  headerContainer: {
+    position: 'relative',
+    overflow: 'hidden',
+    height: 40
+  },
+  headerInner: {
+    position: 'absolute'
+  },
+  scriptLinkWithUpdatedAt: {
+    display: 'block'
+  },
+  outerContainer: {
+    textAlign: 'right'
+  },
+  containerWithUpdatedAt: {
+    verticalAlign: 'bottom',
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    display: 'inline-block'
+  },
+  headerVignetteRight: {
+    position: 'absolute',
+    width: '100%',
+    height: 18,
+    pointerEvents: 'none',
+    background:
+      'linear-gradient(to right, rgba(0, 173, 188, 0) calc(100% - 20px), rgba(0, 173, 188, 1) 100%)'
+  }
+};
+
+export default class HeaderFinish extends React.Component {
+  static propTypes = {
+    lessonData: PropTypes.object,
+    width: PropTypes.number,
+    setDesiredWidth: PropTypes.func
+  };
+
+  setDesiredWidth() {
+    // Report back to our parent how wide we would like to be.
+    const fullWidth = $('.header_finished').width();
```

Comment:
Have you considered using [refs](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) rather than jquery to get this information?

Reply:
I did, and was basically just not sure whether it's as [reliable](https://stackoverflow.com/a/6856375) cross-browser as jQuery.

Reply:
FWIW :thinking: you can have the best of both worlds: Use React refs to get a handle on a DOM element without performing a global DOM query, then wrap that handle in a jQuery object to take advantage of its cross-browser affordances.

Reply:
Good point; I'll do that.